nanome/api/structure/molecule.py

```diff
@@ -165,6 +170,26 @@ def copy_conformer(self, src, index=None):
         self._copy_conformer(src, index)
     # endregion
 
+    def get_substructures(self, callback=None):
+        expects_response = callback is not None or nanome.PluginInstance._instance.is_async
+        id = _ProcessNetwork._send(_Messages.substructure_request, (self.index, nanome.util.enums.SubstructureType.Unkown), expects_response)
+        return nanome.PluginInstance._save_callback(id, callback)
```

Comment:
You need to return a future for async calls, you can look at _shape._upload implementation
I feel like it should be a bit more centralized to make it easier to implement callbacks

Reply:
`nanome.PluginInstance._save_callback` returns a `future`. `_shape._upload` needed a manual thing because it needed to do some post-processing before returning result

Reply:
mb, I totally forgot that